### ISSUE TITLE
False Positive in Analyzer IDE0052 when ref Fields are involved

### ISSUE THREAD

**Issue by fitdev** [dotnet/roslyn#77251]
**Version Used**: 
VS 17.13 Preview 1
DotNet 9
Lang Version = Preview

**Steps to Reproduce**:
```cs
public readonly ref struct RefScope<T> {

  public RefScope(ref T originalvalue, T newvalue) {
    _OriginalValue = originalvalue;
    _Reference = ref originalvalue;
    originalvalue = newvalue;
  }

  readonly ref T _Reference; //<-- IDE0052: Private member 'RefScope._Reference' can be removed as the value assigned to it is never read
  readonly T _OriginalValue;

  public void Dispose() => _Reference = _OriginalValue;

}
```

Here `_Reference` is definitely needed as it is being mutated by a public `Dispose` API, and even though it is never read within the type, effectively its value can be read by the users of the type because it is a `ref` field.

**Diagnostic Id**:

IDE0052: Private member 'RefScope._Reference' can be removed as the value assigned to it is never read

**Expected Behavior**:
`ref` fields that are mutated by _public_ APIs should not be considered safe to remove by IDE0052 Analyzer

**Actual Behavior**:
Analyzer suggests it is safe to remove such a field.

**Comment by RikkiGibson** [comment]
I think we had an issue where a value assignment of a ref field was not considered to be reading the reference. This may have been fixed. Please let us know if this repros in 17.14 preview 1.

**Comment by deepakrathore33** [comment]
I was unable to reproduce this issue in Visual Studio 17.14 Preview 2. It seems the problem may have been resolved. If you continue to experience this issue, please feel free to reopen it.

**Comment by fitdev** [comment]
@deepakrathore33 Just got around to trying ti with the latest VS 2022 17.14 Preview 3, and I still get this false positive. The issue is still there and has not been fixed. Please reopen.

![Image](https://github.com/user-attachments/assets/92139312-b24b-437b-a7e1-52bee14f8693)

**Comment by CyrusNajmabadi** [comment]
I don't repro this either:

![Image](https://github.com/user-attachments/assets/8d7e33d1-37c2-475d-adee-169028fa2b70)

Your UI also seems very strange.  It doesn't look like VS.  Is it possible it's some other tool reporting that?  

Finally, this may be something to do with your project config.  Can you supply a repro project that shows this issue happening?

**Comment by fitdev** [comment]
Here's the repro.

[20250417.IDE0052.repro.zip](https://github.com/user-attachments/files/19799376/20250417.IDE0052.repro.zip)
